### PR TITLE
Code Analysis

### PR DESCRIPTION
**Pull request by wesleybl**
- Black
- Flak8
- isort

### CODE REVIEW

File: src/collective/jsonmigrator/blueprints/datafields.py [hunk 1]
```diff
@@ -27,8 +29,15 @@ def __init__(self, transmogrifier, name, options, previous):
         self.datafield_prefix = options.get("datafield-prefix", "_datafield_")
         self.root_path_length = len(self.context.getPhysicalPath())
 
+        if "path-key" in options:
+            pathkeys = options["path-key"].splitlines()
+        else:
+            pathkeys = defaultKeys(options["blueprint"], name, "path")
+        self.pathkey = Matcher(*pathkeys)
+
```
Comment (mauritsvanrees) [inline src/collective/jsonmigrator/blueprints/datafields.py:37]: Was this supposed to go in here? This does not have anything to do with code analysis.

File: src/collective/jsonmigrator/blueprints/datafields.py [hunk 2]
```diff
@@ -27,8 +29,15 @@ def __init__(self, transmogrifier, name, options, previous):
...
     def __iter__(self):
         for item in self.previous:
+            pathkey = self.pathkey(*list(item.keys()))[0]
```
Comment (mauritsvanrees) [inline src/collective/jsonmigrator/blueprints/datafields.py:40]: Same here: Was this supposed to go in here? This does not have anything to do with code analysis.

**Review by mauritsvanrees — CHANGES_REQUESTED** [review]
Two inline comments, about things that have nothing to do with code analysis, so I wonder if 'real' changes have slipped in unintentionally. But the code looks familiar, so maybe this is supposed to be in every blueprint. I have hardly used transmogrifier, so I don't know. You may know better than me if this is okay.

Other than that, LGTM.

File: src/collective/jsonmigrator/blueprints/datafields.py (hunk 2 above)
Comment (wesleybl) [reply src/collective/jsonmigrator/blueprints/datafields.py:40]: Flake8 complained that the `pathkey` variable was not defined. And it really wasn't:

https://github.com/collective/collective.jsonmigrator/blob/32b58f717e7f01356d9d1e19566c4b2d6cc9ffce/src/collective/jsonmigrator/blueprints/datafields.py#L30-L38

This error was caused by the commit:

https://github.com/collective/collective.jsonmigrator/commit/32b58f717e7f01356d9d1e19566c4b2d6cc9ffce#diff-12d065283fd4ff3aa1d267db4a49c4c83f62d6b841979d9fe6ed11e8256ddeebL38

All blueprints define the `pathkey` variable, except this one.

The definition of this variable, enables to define a custom key in json, to specify the object path. So I decided to add this functionality to this bluiprint as well.

I am not using this blueprint. That's why I didn't notice this error before.

File: src/collective/jsonmigrator/blueprints/datafields.py (hunk 1 above)
Comment (wesleybl) [reply src/collective/jsonmigrator/blueprints/datafields.py:37]: This code was needed to define `pathkey` variable. See:

https://github.com/collective/collective.jsonmigrator/pull/42#discussion_r707636045

File: src/collective/jsonmigrator/blueprints/datafields.py (hunk 2 above)
Comment (wesleybl) [reply src/collective/jsonmigrator/blueprints/datafields.py]: @mauritsvanrees I decided to fix in this PR only the undefined variable. We can add pathkey support in another PR.

Can you please review it again?

Comment (mauritsvanrees) [reply src/collective/jsonmigrator/blueprints/datafields.py]: LGTM

**Review by mauritsvanrees — APPROVED** [review]

**Pull request merged**